File: forest/drop-the-flag.lua
function pos2xy(pos) return pos['x'], pos['y'] end

loop
    local flag = self:findFlag()
    if flag then
        self:pickUpFlag(flag)
        self:buildXY("fire-trap", pos2xy(flag.pos))
    else
        local item = self:findNearestItem()
        if item then
            self:moveXY(pos2xy(item.pos))
        end
    end
end

File: forest/drop-the-flag.clj
(while true
  (def flag (.findFlag this))
  (if flag
    (when true
      (.pickUpFlag this flag)
      (.buildXY this "fire-trap" (.x (.pos flag)) (.y (.pos flag))))
    (when true
      (def item (.findNearest this (.findItems this)))
      (if item
        (.move this (.pos item))
        (.move this (clj->js { :x 20, :y 32 }))))))
